### ISSUE TITLE
OpenAPI schema: views do not indicate required properties of underlying tables

### ISSUE THREAD

**Issue by bnjmnt4n** [PostgREST/postgrest#1947]
### Environment

* PostgreSQL version: 12.5 (running on Supabase)
* PostgREST version: 8.0.0
* Operating system: ?

### Description of issue

When creating a view of a table, the resulting definition in the OpenAPI schema does not have the `required` properties, even if the view does not perform any join operations, in which case the columns returned should have the same `null/not null` requirements as the original table.

Example

```sql
create table customer (
  id uuid primary key not null,
  created_at timestamptz not null default now(),
  updated_at timestamptz not null default now()
)

create view customer_view as
  select * from customer where id = '...';
```

**Comment by wolfgangwalther** [comment]
I'm not sure whether we should do that. There could be `INSTEAD OF` triggers on the view that change the whole logic when inserting or updating a row. Even if the column on the base table would be required (`NOT NULL`) it could have a fallback in the trigger that essentially makes it an optional column.

Although that same argument holds true for the required field on tables, too. There could be `BEFORE` triggers that do the same.

**Comment by bnjmnt4n** [comment]
I agree the behaviour wouldn't be accurate for all use cases, but as you mentioned even tables could have triggers which make the OpenAPI schema inaccurate anyway. Maybe this would be a good default to have? From browsing the issues, I gather there are plans to make the OpenAPI schema an add-on as well, so it's probably fine to revisit when that happens and the schema is made more customizable.

**Comment by sseppola** [comment]
I find this to be a big pain point for code bases that rely on TypeScript.

[From the docs](https://postgrest.org/en/stable/schema_structure.html):
> It is recommended that you don’t expose tables on your API schema. Instead expose views and stored procedures which insulate the internal details from the outside world. This allows you to change the internals of your schema and maintain backwards compatibility. It also keeps your code easier to refactor, and provides a natural way to do API versioning.

This is solid advice which I'm in the process of implementing, however it leaves me having to pick one of:
- type accuracy, developer experience OR
- schema isolation

Schema isolation is a must-have when deploying to a native apps as you'll have a long tail of users using "old queries", so it's crucial that these don't start throwing when the underlying implementation changes. So this leaves us with types like this:
<img width="451" alt="Screenshot 2022-01-01 at 13 35 59" src="https://user-images.githubusercontent.com/4065840/147855490-7eaadfc9-8c1f-42b7-976c-e6a1a7715653.png">

which is very uncomfortable to work with as it provides zero guarantees for what you can rely on.

**Comment by wolfgangwalther** [comment]
We currently have two issues with required fields. One is mentioned in #1653: `required` currently only depends on `NOT NULL` / `NULL`, but it should take into account whether a `DEFAULT` value is set on the column, too.

Once we have implemented that, we can extend this to views and mark columns on views required, iff:
- we can detect the column in the base table for it and this is marked as `NOT NULL` **and**
- the **view** does not contain a `DEFAULT` value for that column

**Comment by shania-g** [comment]
@sseppola Did you find any workaround for this issue? Or did you just avoid schemas completely and directly exposed tables?
